packages/@aws-cdk/aws-cloudfront/README.md

```diff
@@ -103,3 +114,118 @@ const distribution = new CloudFrontWebDistribution(this, 'MyDistribution', {
     ]
 });
 ```
+
+## Distribution API - Experimental
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+In addition to the APIs listed above, a new construct (`Distribution`) is under active development as a newer, friendlier way to define CloudFront
+distributions.
+
+### Creating a distribution
+
+CloudFront distributions deliver your content from one or more origins; an origin is the location where you store the original version of your
+content. Origins can be created from S3 buckets or a custom origin (HTTP server). Each distribution has a default behavior which applies to all
+requests to that distribution, and routes requests to a primary origin.
+
+#### From an S3 Bucket
+
+An S3 bucket can be added as an origin. If the bucket is configured as a website endpoint, the distribution can use S3 redirects and S3 custom error
+documents.
+
+```ts
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+
+// Creates a distribution for a S3 bucket.
+const myBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromBucket(myBucket) },
+});
+
+// Equivalent to the above
+const myBucket = new s3.Bucket(...);
+cloudfront.Distribution.forBucket(this, 'myDist', myBucket);
+
+// Creates a distribution for a S3 bucket that has been configured for website hosting.
+const myWebsiteBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromWebsiteBucket(myBucket) },
+});
+
+// Equivalent to the above
+const myBucket = new s3.Bucket(...);
+cloudfront.Distribution.forWebsiteBucket(this, 'myDist', myBucket);
+```
+
+The `forBucket` options will automatically create an origin access identity and grant it access to the underlying bucket. This can be used in
+conjunction with a bucket that is not public to require that your users access your content using CloudFront URLs and not S3 URLs directly.
+
+### Domain Names and Certificates
+
+When you create a distribution, CloudFront returns a domain name for the distribution, for example: `d111111abcdef8.cloudfront.net`. CloudFront
```

Comment:
Returns how?

Reply:
"Returns" is maybe the wrong word. It's assigned by CloudFront, and returned (by CloudFormation) as an available attribute (`domainName`) from Fn::GetAtt. Within context, I think `assigns` is better.

```suggestion
When you create a distribution, CloudFront assigns a domain name for the distribution, for example: `d111111abcdef8.cloudfront.net`. CloudFront
```

Reply:
I would also mention that it can be obtained through `distribution.domainName`